### PR TITLE
WT-5698 Change assert from WT_ERR to WT_RET

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

**Comment by luke-pearson** [comment]
It just needs to be below the __cursor_state_save(cursor, &state); call, as it can jump to err on non diagnostic builds.

File: src/btree/bt_cursor.c [hunk 1]
```diff
@@ -805,8 +805,6 @@ __wt_btcur_insert(WT_CURSOR_BTREE *cbt)
 
     /* It's no longer possible to bulk-load into the tree. */
     __wt_cursor_disable_bulk(session);
-    WT_ERR_ASSERT(
```
Comment (agorrod) [inline src/btree/bt_cursor.c]: The better change here is to switch to WT_RET_ASSERT, and I think it should go above the disable_bulk call - that's the code that assumes the session holds the "right" btree. Similarly below.

I'm happy to talk about when to choose ERR vs RET if that would help.

Comment (luke-pearson) [reply src/btree/bt_cursor.c]: No that makes sense.

**Comment by luke-pearson** [comment]
evergreen retry

**Pull request merged**